ui/src/providers/RootProvider.js

```diff
@@ -18,13 +18,15 @@ const arrangerFetcher = async (args) => {
 };
 
 const RootProvider = ({ children }) => (
-  <SelectedModelsProvider>
-    <ArrangerDataProvider documentType={'model'} customFetcher={arrangerFetcher}>
-      <SavedSetsProvider>
-        <ModalStateProvider>{children}</ModalStateProvider>
-      </SavedSetsProvider>
-    </ArrangerDataProvider>
-  </SelectedModelsProvider>
+  <ArrangerDataProvider documentType={'model'} customFetcher={arrangerFetcher}>
+    <TableContextProvider>
+      <SelectedModelsProvider>
+        <SavedSetsProvider>
+          <ModalStateProvider>{children}</ModalStateProvider>
+        </SavedSetsProvider>
+      </SelectedModelsProvider>
+    </TableContextProvider>
+  </ArrangerDataProvider>
```

Comment:
Aha! I commented on this provider order in the base PR, good to see you're already a step ahead 😛

Reply:
Yeah there's a lot of progressive improvement from branch to branch, hope it's not too confusing